caregiver_application/web_application/templates/workflow.html

```diff
@@ -121,46 +112,156 @@
                 <button type="submit" id="btnSubmitObservation" class="btn btn-primary btn-md">Submit</button>
                 <input type="hidden" value="{{ patId }}" name="patientId" />
                 <input type="hidden" id="hidValueType" value="" name="obervationValueType" />
-                
+
             </div>
         </form>
     </div>
 </div>
 <script>
-    $(function () {        
-        var currentTaskId = 0;
-        $(".task-info .observation-opener span").click(function () {
-            currentTaskId = $(this).attr("data-taskId");
-            var action = '/workflow/' + {{ id }} + '/' + currentTaskId + '/observe';
-            $("#frmObservation").attr("action",action);
-            $(".modal-wrapper").fadeIn();
-        });
-        $(".modal-wrapper .close").click(function () {
-            $(".modal-wrapper").fadeOut();
-        });       
-
-      var observeDone = getUrlVars()["observeDone"];
-      if(observeDone == "True")
-      {        
-        $(".alert").slideToggle();
-        hideAlert();
-      }    
-      else if(observeDone!== undefined)
-        alert("Adding observation failed!");
-
-        function hideAlert(){
+    document.addEventListener('DOMContentLoaded', function () {
+        const workflowId = '{{ id }}';
+        const patientId = '{{ patId }}';
+        const observationCodeSelect = document.getElementById('observationCode');
+        const observationValueInput = document.getElementById('observationValue');
+        const modalWrapper = document.querySelector('.modal-wrapper');
+        const alertBox = document.querySelector('.alert');
+
+        // Function to hide alert box after 3 seconds
+        function hideAlert() {
             setTimeout(() => {
-            $(".alert").slideToggle()
-        }, 3000);
+                if (alertBox) {
+                    alertBox.style.display = 'none';
+                }
+            }, 3000);
         }
 
-        $('#observationCode').change(()=>{
-            var dataType = $('#observationCode :selected').attr('data-type');
-            $('#hidValueType').val(dataType);
-        });       
+        // Initialize event listeners
+        function init() {

```

Comment:
very generic name for a function that is not on the top of the file. Maybe initEvents() ?